groovy-language-server/src/main/java/com/palantir/groovylanguageserver/GroovycWrapper.java

```diff
@@ -176,31 +227,61 @@ private void addAllSourcesToCompilationUnit() {
 
     private void parseAllSymbols() {
         Map<String, Set<SymbolInformation>> newFileSymbols = Maps.newHashMap();
+        Map<String, Set<SymbolInformation>> newTypeReferences = Maps.newHashMap();
+
         unit.iterator().forEachRemaining(sourceUnit -> {
             Set<SymbolInformation> symbols = Sets.newHashSet();
             String sourcePath = sourceUnit.getSource().getURI().getPath();
+
             // This will iterate through all classes, interfaces and enums, including inner ones.
             sourceUnit.getAST().getClasses().forEach(clazz -> {
                 // Add class symbol
-                symbols.add(createSymbolInformation(clazz.getName(), getKind(clazz), createLocation(sourcePath, clazz),
-                        Optional.fromNullable(clazz.getOuterClass()).transform(ClassNode::getName)));
+                SymbolInformation classSymbol =
+                        createSymbolInformation(clazz.getName(), getKind(clazz), createLocation(sourcePath, clazz),
+                                Optional.fromNullable(clazz.getOuterClass()).transform(ClassNode::getName));
+                symbols.add(classSymbol);
+
+                // Add implemented interfaces reference
+                for (ClassNode node : clazz.getInterfaces()) {
```

Comment:
prefer for this to be in java 8 to reduce if/else and also to match the rest of this method:
```
                // Add implemented interfaces reference
                Stream.of(clazz.getInterfaces())
                        .filter(node -> !node.getName().equals(GROOVY_DEFAULT_INTERFACE))
                        .forEach(node -> addToValueSet(newTypeReferences, node.getName(), classSymbol));
```